libs/agno/agno/workflow/v2/task.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Type, Union
+from uuid import uuid4
+
+from agno.agent import Agent
+from agno.run.response import RunResponse
+from agno.team import Team
+from agno.utils.log import logger
+
+
+@dataclass
+class Task:
+    """A single unit of work in a workflow sequence"""
+
+    name: str
+    executor: Union[Agent, Team]
+
+    task_id: Optional[str] = None
+    description: Optional[str] = None
+    expected_input: Optional[Dict[str, Type]] = None
+    expected_output: Optional[str] = None
+
+    # Task configuration
+    retry_count: int = 0
+    max_retries: int = 3
+    timeout_seconds: Optional[int] = None
+
+    skip_on_failure: bool = False
+
+    # Input validation mode
+    # If False, only warn about missing inputs
+    strict_input_validation: bool = False
+
+    def __post_init__(self):
+        if self.task_id is None:
+            self.task_id = str(uuid4())
+
+    def execute(self, inputs: Dict[str, Any], context: Dict[str, Any] = None) -> RunResponse:
+        """Execute the task with given inputs synchronously"""
+        logger.info(f"Executing task: {self.name}")
+
+        # Validate inputs if expected_input is defined
+        if self.expected_input:
+            validation_result = self._validate_inputs(inputs)
+            if not validation_result and self.strict_input_validation:
+                raise ValueError(f"Input validation failed for task {self.name}")
+
+        # Execute with retries
+        for attempt in range(self.max_retries + 1):
+            try:
+                if isinstance(self.executor, Agent):
+                    # Format inputs for agent
+                    message = self._format_inputs_for_agent(inputs)
+                    response = self.executor.run(message)
```

Comment:
you now have to pass message, image, etc.
You'll have to "initiliaze" the member with context and workflow_session_state

Remember to make the agent/team output the input of the next.
Remember to deal with pydantic output that becomes input